index.js

```diff
@@ -277,9 +282,45 @@ app.post('/notify', function(req, res) {
       throw new Error('Not Found');
     }
 
+    // check activity and sendNotification if not specified or "1"
+    function checkActivity(token, machine, client, registration, payload, ttl) {
+      // XXX: this should be a hash per machine
+      var machineKey = token + ':' + machine;
+      return redis.hget(machineKey, client)
+      .then(function(isActive) {
+        if (isActive === '0') {
+          // not sending if token:machine client is "0"
+          return;
+        }
+        if (isActive === '1') {
+          // client is active on this machine
+          return sendNotification(token, registration, payload, ttl);
+        }
+        return redis.hmset(machineKey, client, 1)
+        .then(function() {
+          // adding a client to the machine, '1' by default
+          // check for multiple clients
+          return redis.sismember(token + ':clients', client)
+          .then(function(isMember) {
+            return new Promise(function(resolve, reject) {
```

Comment:
I could do this:             if (!isMember) {
              return redis.sadd(token + ':clients', client)
              .then(function() {
                return sendNotification(token, registration, payload, ttl);
              });
            }
            return sendNotification(token, registration, payload, ttl);